modules/dnn/test/test_caffe_importer.cpp

```diff
@@ -667,8 +699,17 @@ TEST_P(Test_Caffe_nets, RFCN)
         applyTestTag(CV_TEST_TAG_DNN_SKIP_IE_OPENCL_FP16);
     if (backend == DNN_BACKEND_INFERENCE_ENGINE && target == DNN_TARGET_MYRIAD)
         applyTestTag(CV_TEST_TAG_DNN_SKIP_IE_MYRIAD);
-    double scoreDiff = (backend == DNN_BACKEND_OPENCV && target == DNN_TARGET_OPENCL_FP16) ? 4e-3 : default_l1;
-    double iouDiff = (backend == DNN_BACKEND_OPENCV && target == DNN_TARGET_OPENCL_FP16) ? 8e-2 : default_lInf;
+    float scoreDiff = default_l1, iouDiff = default_lInf;
+    if (backend == DNN_BACKEND_OPENCV && target == DNN_TARGET_OPENCL_FP16)
+    {
+        scoreDiff = 4e-3;
+        iouDiff = 8e-2;
+    }
+    if (target == DNN_TARGET_CUDA_FP16)
+    {
+        scoreDiff = 0.04;
+        iouDiff = 0.11;
```

Comment:
Please add a comment about the range of reference output values. It can help t understand if this threshold is acceptable or not.

Reply:
The ranges for both scores and IOU is `[0, 1]`.

The thresholds have been updated:
```
scoreDiff = 0.0034;
iouDiff = 0.11;
```

The thresholds used by OCL_FP16 and MYRAID:
```
scoreDiff = 4e-3;
iouDiff = 8e-2;
```

The iouDiff threshold of `0.08` is close to `0.11`. I think there is no bug with CUDA_FP16 and the thresholds are correct.